compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -148,11 +148,11 @@ where
 /// converted rather than on `DiagArgValue`, which enables types from other `rustc_*` crates to
 /// implement this.
 pub trait IntoDiagArg {
-    fn into_diag_arg(self) -> DiagArgValue;
+    fn into_diag_arg(self, path: &mut Option<std::path::PathBuf>) -> DiagArgValue;
```

Comment:
Please add a doc comment explaining the meaning of this argument.

Reply:
Added to the trait docstring

Reply:
That won't show up when hovering a method call though, will it? Seems odd to document a method parameter on the trait level.

Reply:
Moved to the method. Any comments on the wording?